_blogposts/2023-05-25-dynamic-import.mdx

```diff
@@ -0,0 +1,138 @@
+---
+author: rescript-team
+date: "2023-05-25"
+title: Dynamic import
+badge: roadmap
+description: |
+  A tour of new capabilities coming in ReScript v11
+---
+
+> This is the third post covering new capabilities that'll ship in ReScript v11. You can check out the first post on [better interop with customizable variants here](https://rescript-lang.org/blog/improving-interop) and the second one TODO.
+
+[Dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import), a feature allowing for asynchronous code loading in JavaScript, has long offered potential for code splitting, lazy loading, and via that reduced initial load times for applications. While ReScript has always supported dynamic imports, they've often been cumbersome to maintain in their current form. We're excited to announce that, with ReScript v11, we're introducing first-class support for dynamic imports - making them ergonomic to use and easy to maintain.
+
+But before we dive into how dynamic imports look in v11, let's do a short primer on what the challenges has been when using dynamic imports in ReScript prior to v11.
+
+## The current state of dynamic imports in ReScript
+
+Dynamic imports in JavaScript work on _paths_. This means that if you want to dynamically import the file `src/utils/text/markdown.js` from `src/components/text/MarkdownRenderer.js`, you'd need to spell the full relative path out:
+
+```js
+// MarkdownRenderer.js
+let markdown = await import("../../../utils/text/markdown.js");
+```
+
+In JavaScript, modules and files are referred to by their file system path, which makes sense considering its design. However, in ReScript, all files represent modules that are globally available for anyone to use, without having to care about where that file is located on the file system.
+
+This global availability of modules is generally beneficial, as it simplifies file restructuring and relieves developers from the need to remember the precise location of each file. However, when it comes to dynamic imports, this poses a challenge.
+
+As a ReScript developer, you're not accustomed to tracking the relative file system locations of your modules. But with dynamic imports, you'd would have to manually keep track of where your files are located relative to each other, in order for your dynamic import binding to point to the right file.
```

Comment:
```suggestion
As a ReScript developer, you're not accustomed to tracking the relative file system locations of your modules. But with dynamic imports, you would have to manually keep track of where your files are located relative to each other, in order for your dynamic import binding to point to the right file.
```

Reply:
Reduced this into a one-liner in the introduction.